internal/services/systemcentervirtualmachinemanager/system_center_virtual_machine_manager_inventory_items_data_source.go

```diff
@@ -0,0 +1,159 @@
+	"github.com/hashicorp/go-azure-sdk/resource-manager/systemcentervirtualmachinemanager/2023-10-07/vmmservers"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+)
+
+type SystemCenterVirtualMachineManagerInventoryItemsDataSource struct{}
+
+var _ sdk.DataSource = SystemCenterVirtualMachineManagerInventoryItemsDataSource{}
+
+type SystemCenterVirtualMachineManagerInventoryItemsDataSourceModel struct {
+	SystemCenterVirtualMachineManagerServerId string          `tfschema:"system_center_virtual_machine_manager_server_id"`
+	InventoryItems                            []InventoryItem `tfschema:"inventory_items"`
+}
+
+type InventoryItem struct {
+	id            string `tfschema:"id"`
+	name          string `tfschema:"name"`
+	InventoryType string `tfschema:"inventory_type"`
+	Uuid          string `tfschema:"uuid"`
+}
+
+func (l SystemCenterVirtualMachineManagerInventoryItemsDataSource) ResourceType() string {
+	return "azurerm_system_center_virtual_machine_manager_inventory_items"
+}
+
+func (l SystemCenterVirtualMachineManagerInventoryItemsDataSource) ModelObject() interface{} {
+	return &SystemCenterVirtualMachineManagerInventoryItemsDataSourceModel{}
+}
+
+func (l SystemCenterVirtualMachineManagerInventoryItemsDataSource) Arguments() map[string]*schema.Schema {
+	return map[string]*pluginsdk.Schema{
+		"system_center_virtual_machine_manager_server_id": commonschema.ResourceIDReferenceRequired(&vmmservers.VMmServerId{}),
+	}
+}
+
+func (l SystemCenterVirtualMachineManagerInventoryItemsDataSource) Attributes() map[string]*schema.Schema {
+	return map[string]*pluginsdk.Schema{
+		"inventory_items": {
+			Type:     pluginsdk.TypeList,
+			Computed: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"id": {
+						Type:     pluginsdk.TypeString,
+						Computed: true,
+					},
+
+					"name": {
+						Type:     pluginsdk.TypeString,
+						Computed: true,
+					},
+
+					"inventory_type": {
+						Type:     pluginsdk.TypeString,
+						Computed: true,
+					},
+
+					"uuid": {
+						Type:     pluginsdk.TypeString,
+						Computed: true,
+					},
+				},
+			},
+		},
+	}
+}
+
+func (l SystemCenterVirtualMachineManagerInventoryItemsDataSource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.SystemCenterVirtualMachineManager.InventoryItems
+
+			var state SystemCenterVirtualMachineManagerInventoryItemsDataSourceModel
+			if err := metadata.Decode(&state); err != nil {
+				return err
+			}
+
+			scvmmServerId, err := inventoryitems.ParseVMmServerIDInsensitively(state.SystemCenterVirtualMachineManagerServerId)
+			if err != nil {
+				return err
+			}
+
+			id := fmt.Sprintf("%s/inventoryItems", scvmmServerId.ID())
```

Comment:
Why aren't we constructing this ID with `inventoryitems.NewInventoryItemID()`?

Reply:
Reason 1: `inventoryitems.NewInventoryItemID()` points to the dependent Inventory Item resource and service team expects us to not expose it to users.

Reason 2: Setting `inventoryitems.NewInventoryItemID()` requires the specified Inventory Item Id but this data source is used to expose a list of Inventory Items. So I assume that we couldn't set it with any Inventory Item Id in the list.

Reply:
We should not be defining Terraform specific Resource IDs where they aren't _absolutely_ necessary - this is a great example of where we should be using the SCVMM Server ID rather than creating something random - can we update this as @stephybun mentions

Reply:
Updated to use SCVMM Server ID.